### ISSUE TITLE
[Bug] Error when using CCIT compression

### ISSUE THREAD

**Issue by MKH-42** [gotson/komga#377]
### Komga environment
- **OS**: Windows
- **Komga version**:  komga-0.64.8
- [ ] **I am running Komga with Docker**
  - Docker image tag [e.g. latest, beta]: 
- [X] **I am running Komga from the `jar`**
  - Java version:  1.8.0_271
- [ ] **I have a problem in the web interface**
  - Browser (with version): 
- [ ] **I have a problem with an OPDS client application**
  - OPDS Application (with version): 
- [ ] **I have a problem with the Tachiyomi extension**
  - Tachiyomi version: 
  - Tachiyomi extension version: 

### Describe the bug
During scanning a comic book the analyzer throws following error:
```
javax.imageio.IIOException: No SOF segment in stream
	at com.twelvemonkeys.imageio.plugins.jpeg.JPEGImageReader.getSOF(JPEGImageReader.java:888) ~[imageio-jpeg-3.6.1.jar!/:3.6.1]
```

### Steps to reproduce
Scan a JPEG image with TIFF compression 

### Expected behavior
Scan should work like all other scans.

### Actual behavior
Comic book  can not analyzied. 

### Additional context
It looks like the error comes from twelvemonkeys modul.
See also https://github.com/haraldk/TwelveMonkeys/commit/253f04066bbafc397dd6278d531eee9569554c26 

### Log file
If applicable, add an excerpt of the log file (max 20 lines), or a link to a gist containing the log file.
```
javax.imageio.IIOException: No SOF segment in stream
	at com.twelvemonkeys.imageio.plugins.jpeg.JPEGImageReader.getSOF(JPEGImageReader.java:888) ~[imageio-jpeg-3.6.1.jar!/:3.6.1]
	at com.twelvemonkeys.imageio.plugins.jpeg.JPEGImageReader.getWidth(JPEGImageReader.java:215) ~[imageio-jpeg-3.6.1.jar!/:3.6.1]
	at org.gotson.komga.infrastructure.image.ImageAnalyzer.getDimension(ImageAnalyzer.kt:21) ~[classes!/:na]
	at org.gotson.komga.infrastructure.mediacontainer.RarExtractor.getEntries(RarExtractor.kt:41) ~[classes!/:na]
	at org.gotson.komga.domain.service.BookAnalyzer.analyze(BookAnalyzer.kt:42) [classes!/:na]
	at org.gotson.komga.domain.service.BookLifecycle.analyzeAndPersist(BookLifecycle.kt:42) [classes!/:na]
	at org.gotson.komga.application.tasks.TaskHandler.handleTask(TaskHandler.kt:44) [classes!/:na]
	at sun.reflect.GeneratedMethodAccessor205.invoke(Unknown Source) ~[na:na]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[na:1.8.0_271]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[na:1.8.0_271]
	at org.springframework.messaging.handler.invocation.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:171) [spring-messaging-5.3.2.jar!/:5.3.2]
	at org.springframework.messaging.handler.invocation.InvocableHandlerMethod.invoke(InvocableHandlerMethod.java:120) [spring-messaging-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.adapter.MessagingMessageListenerAdapter.invokeHandler(MessagingMessageListenerAdapter.java:114) [spring-jms-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.adapter.MessagingMessageListenerAdapter.onMessage(MessagingMessageListenerAdapter.java:77) [spring-jms-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.AbstractMessageListenerContainer.doInvokeListener(AbstractMessageListenerContainer.java:736) [spring-jms-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.AbstractMessageListenerContainer.invokeListener(AbstractMessageListenerContainer.java:696) [spring-jms-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.AbstractMessageListenerContainer.doExecuteListener(AbstractMessageListenerContainer.java:674) [spring-jms-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.AbstractPollingMessageListenerContainer.doReceiveAndExecute(AbstractPollingMessageListenerContainer.java:318) [spring-jms-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.AbstractPollingMessageListenerContainer.receiveAndExecute(AbstractPollingMessageListenerContainer.java:257) [spring-jms-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.DefaultMessageListenerContainer$AsyncMessageListenerInvoker.invokeListener(DefaultMessageListenerContainer.java:1189) [spring-jms-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.DefaultMessageListenerContainer$AsyncMessageListenerInvoker.executeOngoingLoop(DefaultMessageListenerContainer.java:1179) [spring-jms-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.DefaultMessageListenerContainer$AsyncMessageListenerInvoker.run(DefaultMessageListenerContainer.java:1076) [spring-jms-5.3.2.jar!/:5.3.2]
	at java.lang.Thread.run(Unknown Source) [na:1.8.0_271]
```

**Comment by MKH-42** [comment]
[01.zip](https://github.com/gotson/komga/files/5767271/01.zip)
Example

**Comment by gotson** [comment]
I cannot reproduce the problem, it works on my end and the cover is generated properly.

<img width="534" alt="Screenshot 2021-01-05 at 09 14 40" src="https://user-images.githubusercontent.com/2139133/103595479-797d1180-4f36-11eb-8b8b-a5c67e84252e.png">

**Comment by gotson** [comment]
I see in your log excerpt that the file is a RAR, but here you gave me a ZIP. I saw this in the past, where the RAR is partly corrupt, and when extracting it would generate those issues.

Can you convert your book to ZIP and try?

**Comment by MKH-42** [comment]
Sorry, was the wrong file.
Here the right on.


[01.zip](https://github.com/gotson/komga/files/5770423/01.zip)

**Comment by gotson** [comment]
Can you provide the original rar instead?

**Comment by MKH-42** [comment]
[TEST.zip](https://github.com/gotson/komga/files/5770459/TEST.zip)
Intersting: CBZ works, CBR not.

**Comment by MKH-42** [comment]
I put now in a zip 2 testfiles .. a zip and a rar version ..

**Comment by MKH-42** [comment]
![Screenshot](https://user-images.githubusercontent.com/76835223/103653521-eeffd680-4f64-11eb-8855-05d4220888bd.JPG)

**Comment by MKH-42** [comment]
2021-01-05 14:43:55.310  INFO 10216 --- [DefaultMessageListenerContainer-1] o.g.k.application.tasks.TaskReceiver     : Sending task: GenerateBookThumbnail(bookId=03Q8R1K9K8BHR)
2021-01-05 14:43:55.310  INFO 10216 --- [DefaultMessageListenerContainer-1] o.g.k.application.tasks.TaskReceiver     : Sending task: RefreshBookMetadata(bookId=03Q8R1K9K8BHR)
2021-01-05 14:43:55.310  INFO 10216 --- [DefaultMessageListenerContainer-1] o.g.komga.application.tasks.TaskHandler  : Task AnalyzeBook(bookId=03Q8R1K9K8BHR) executed in 96.3ms
2021-01-05 14:43:55.388  INFO 10216 --- [DefaultMessageListenerContainer-1] o.g.komga.application.tasks.TaskHandler  : Executing task: AnalyzeBook(bookId=03Q8RJNNZ87SK)
2021-01-05 14:43:55.388  INFO 10216 --- [DefaultMessageListenerContainer-1] o.g.komga.domain.service.BookLifecycle   : Analyze and persist book: Book(name=rartest, url=file:/V:/Komga/TEST/rartest.cbr, fileLastModified=2021-01-05T14:41:41.341, fileSize=1818925, number=2, id=03Q8RJNNZ87SK, seriesId=03Q6M9TAZ88R6, libraryId=03Q6ENAYF88XV, createdDate=2021-01-05T14:42:52, lastModifiedDate=2021-01-05T14:42:52.778)
2021-01-05 14:43:55.403  INFO 10216 --- [DefaultMessageListenerContainer-1] o.g.komga.domain.service.BookAnalyzer    : Trying to analyze book: Book(name=rartest, url=file:/V:/Komga/TEST/rartest.cbr, fileLastModified=2021-01-05T14:41:41.341, fileSize=1818925, number=2, id=03Q8RJNNZ87SK, seriesId=03Q6M9TAZ88R6, libraryId=03Q6ENAYF88XV, createdDate=2021-01-05T14:42:52, lastModifiedDate=2021-01-05T14:42:52.778)
2021-01-05 14:43:55.403  INFO 10216 --- [DefaultMessageListenerContainer-1] o.g.komga.domain.service.BookAnalyzer    : Detected media type: application/x-rar-compressed; version=4
2021-01-05 14:43:55.482  WARN 10216 --- [DefaultMessageListenerContainer-1] o.g.k.i.mediacontainer.RarExtractor      : Could not analyze entry: 01.jpg

javax.imageio.IIOException: No SOF segment in stream
	at com.twelvemonkeys.imageio.plugins.jpeg.JPEGImageReader.getSOF(JPEGImageReader.java:888) ~[imageio-jpeg-3.6.1.jar!/:3.6.1]
	at com.twelvemonkeys.imageio.plugins.jpeg.JPEGImageReader.getWidth(JPEGImageReader.java:215) ~[imageio-jpeg-3.6.1.jar!/:3.6.1]
	at org.gotson.komga.infrastructure.image.ImageAnalyzer.getDimension(ImageAnalyzer.kt:21) ~[classes!/:na]
	at org.gotson.komga.infrastructure.mediacontainer.RarExtractor.getEntries(RarExtractor.kt:41) ~[classes!/:na]
	at org.gotson.komga.domain.service.BookAnalyzer.analyze(BookAnalyzer.kt:42) [classes!/:na]
	at org.gotson.komga.domain.service.BookLifecycle.analyzeAndPersist(BookLifecycle.kt:42) [classes!/:na]
	at org.gotson.komga.application.tasks.TaskHandler.handleTask(TaskHandler.kt:44) [classes!/:na]
	at sun.reflect.GeneratedMethodAccessor225.invoke(Unknown Source) ~[na:na]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[na:1.8.0_271]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[na:1.8.0_271]
	at org.springframework.messaging.handler.invocation.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:171) [spring-messaging-5.3.2.jar!/:5.3.2]
	at org.springframework.messaging.handler.invocation.InvocableHandlerMethod.invoke(InvocableHandlerMethod.java:120) [spring-messaging-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.adapter.MessagingMessageListenerAdapter.invokeHandler(MessagingMessageListenerAdapter.java:114) [spring-jms-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.adapter.MessagingMessageListenerAdapter.onMessage(MessagingMessageListenerAdapter.java:77) [spring-jms-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.AbstractMessageListenerContainer.doInvokeListener(AbstractMessageListenerContainer.java:736) [spring-jms-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.AbstractMessageListenerContainer.invokeListener(AbstractMessageListenerContainer.java:696) [spring-jms-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.AbstractMessageListenerContainer.doExecuteListener(AbstractMessageListenerContainer.java:674) [spring-jms-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.AbstractPollingMessageListenerContainer.doReceiveAndExecute(AbstractPollingMessageListenerContainer.java:318) [spring-jms-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.AbstractPollingMessageListenerContainer.receiveAndExecute(AbstractPollingMessageListenerContainer.java:257) [spring-jms-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.DefaultMessageListenerContainer$AsyncMessageListenerInvoker.invokeListener(DefaultMessageListenerContainer.java:1189) [spring-jms-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.DefaultMessageListenerContainer$AsyncMessageListenerInvoker.executeOngoingLoop(DefaultMessageListenerContainer.java:1179) [spring-jms-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.DefaultMessageListenerContainer$AsyncMessageListenerInvoker.run(DefaultMessageListenerContainer.java:1076) [spring-jms-5.3.2.jar!/:5.3.2]
	at java.lang.Thread.run(Unknown Source) [na:1.8.0_271]

2021-01-05 14:43:55.482  WARN 10216 --- [DefaultMessageListenerContainer-1] o.g.komga.domain.service.BookAnalyzer    : Book Book(name=rartest, url=file:/V:/Komga/TEST/rartest.cbr, fileLastModified=2021-01-05T14:41:41.341, fileSize=1818925, number=2, id=03Q8RJNNZ87SK, seriesId=03Q6M9TAZ88R6, libraryId=03Q6ENAYF88XV, createdDate=2021-01-05T14:42:52, lastModifiedDate=2021-01-05T14:42:52.778) does not contain any pages
2021-01-05 14:43:56.466  INFO 10216 --- [DefaultMessageListenerContainer-1] o.g.k.application.tasks.TaskReceiver     : Sending task: GenerateBookThumbnail(bookId=03Q8RJNNZ87SK)
2021-01-05 14:43:56.482  INFO 10216 --- [DefaultMessageListenerContainer-1] o.g.k.application.tasks.TaskReceiver     : Sending task: RefreshBookMetadata(bookId=03Q8RJNNZ87SK)
2021-01-05 14:43:56.482  INFO 10216 --- [DefaultMessageListenerContainer-1] o.g.komga.application.tasks.TaskHandler  : Task AnalyzeBook(bookId=03Q8RJNNZ87SK) executed in 1.09s
2021-01-05 14:43:56.529  INFO 10216 --- [DefaultMessageListenerContainer-1] o.g.komga.application.tasks.TaskHandler  : Executing task: GenerateBookThumbnail(bookId=03Q8R1K9K8BHQ)
2021-01-05 14:43:56.529  INFO 10216 --- [DefaultMessageListenerContainer-1] o.g.komga.domain.service.BookLifecycle   : Generate thumbnail and persist for book: Book(name=01, url=file:/V:/Komga/TEST/01.zip, fileLastModified=2021-01-05T09:14:15.294, fileSize=1818925, number=1, id=03Q8R1K9K8BHQ, seriesId=03Q6M9TAZ88R6, libraryId=03Q6ENAYF88XV, createdDate=2021-01-05T14:40:32, lastModifiedDate=2021-01-05T14:42:52.778)
2021-01-05 14:43:56.529  INFO 10216 --- [DefaultMessageListenerContainer-1] o.g.komga.domain.service.BookAnalyzer    : Generate thumbnail for book: Book(name=01, url=file:/V:/Komga/TEST/01.zip, fileLastModified=2021-01-05T09:14:15.294, fileSize=1818925, number=1, id=03Q8R1K9K8BHQ, seriesId=03Q6M9TAZ88R6, libraryId=03Q6ENAYF88XV, createdDate=2021-01-05T14:40:32, lastModifiedDate=2021-01-05T14:42:52.778)
2021-01-05 14:43:56.529  WARN 10216 --- [DefaultMessageListenerContainer-1] o.g.komga.domain.service.BookAnalyzer    : Book media is not ready, cannot generate thumbnail. Book: Book(name=01, url=file:/V:/Komga/TEST/01.zip, fileLastModified=2021-01-05T09:14:15.294, fileSize=1818925, number=1, id=03Q8R1K9K8BHQ, seriesId=03Q6M9TAZ88R6, libraryId=03Q6ENAYF88XV, createdDate=2021-01-05T14:40:32, lastModifiedDate=2021-01-05T14:42:52.778)
2021-01-05 14:43:56.529 ERROR 10216 --- [DefaultMessageListenerContainer-1] o.g.komga.domain.service.BookLifecycle   : Error while creating thumbnail

org.gotson.komga.domain.model.MediaNotReadyException: null
	at org.gotson.komga.domain.service.BookAnalyzer.generateThumbnail(BookAnalyzer.kt:85) ~[classes!/:na]
	at org.gotson.komga.domain.service.BookLifecycle.generateThumbnailAndPersist(BookLifecycle.kt:60) ~[classes!/:na]
	at org.gotson.komga.application.tasks.TaskHandler.handleTask(TaskHandler.kt:51) [classes!/:na]
	at sun.reflect.GeneratedMethodAccessor225.invoke(Unknown Source) ~[na:na]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[na:1.8.0_271]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[na:1.8.0_271]
	at org.springframework.messaging.handler.invocation.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:171) [spring-messaging-5.3.2.jar!/:5.3.2]
	at org.springframework.messaging.handler.invocation.InvocableHandlerMethod.invoke(InvocableHandlerMethod.java:120) [spring-messaging-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.adapter.MessagingMessageListenerAdapter.invokeHandler(MessagingMessageListenerAdapter.java:114) [spring-jms-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.adapter.MessagingMessageListenerAdapter.onMessage(MessagingMessageListenerAdapter.java:77) [spring-jms-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.AbstractMessageListenerContainer.doInvokeListener(AbstractMessageListenerContainer.java:736) [spring-jms-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.AbstractMessageListenerContainer.invokeListener(AbstractMessageListenerContainer.java:696) [spring-jms-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.AbstractMessageListenerContainer.doExecuteListener(AbstractMessageListenerContainer.java:674) [spring-jms-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.AbstractPollingMessageListenerContainer.doReceiveAndExecute(AbstractPollingMessageListenerContainer.java:318) [spring-jms-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.AbstractPollingMessageListenerContainer.receiveAndExecute(AbstractPollingMessageListenerContainer.java:257) [spring-jms-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.DefaultMessageListenerContainer$AsyncMessageListenerInvoker.invokeListener(DefaultMessageListenerContainer.java:1189) [spring-jms-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.DefaultMessageListenerContainer$AsyncMessageListenerInvoker.executeOngoingLoop(DefaultMessageListenerContainer.java:1179) [spring-jms-5.3.2.jar!/:5.3.2]
	at org.springframework.jms.listener.DefaultMessageListenerContainer$AsyncMessageListenerInvoker.run(DefaultMessageListenerContainer.java:1076) [spring-jms-5.3.2.jar!/:5.3.2]
	at java.lang.Thread.run(Unknown Source) [na:1.8.0_271]

**Comment by MKH-42** [comment]
Thank you :-)

**Comment by gotson** [comment]
FYI i tried the latest release of [TwelveMonkeys](https://github.com/haraldk/TwelveMonkeys/releases/tag/twelvemonkeys-3.6.2) including the CCIT detection, but that doesn't change the behaviour of Komga when getting the files from a RAR.

**Comment by gotson** [comment]
I think this should be fixed in `0.90.1`
